src/marten/db/errors.cr

```diff
@@ -4,6 +4,9 @@ module Marten
       # Represents an error raised when a specific predicate will return an empty result set.
       class EmptyResults < Exception; end
 
+      # Represents an error raised when using incompatible query sets.
+      class MismatchedQuerySetType < Exception; end
```

Comment:
I believe we can avoid introducing a new error type and we can use the existing `UnmetQuerySetCondition` exception class instead.